### PR TITLE
Fix accidental config breakage of `quarkus.package.decompiler.*` properties

### LINKED ISSUE #40272: Config: reflect a breaking change of package config in the docs
### Description

https://github.com/quarkusio/quarkus/pull/39295 introduced a breaking change that is neither reflected in the docs nor mentioned in the migration guide. The `DecompilerConfig` was moved to `JarConfig` group; as a result the `quarkus.package.decompiler.enabled` property that is used in the [docs](https://quarkus.io/version/main/guides/writing-extensions#troubleshooting-debugging-tips) does not work anymore. The current property name is `quarkus.package.jar.decompiler.enabled`. The same applies to `quarkus.package.decompiler.jar-directory`. I did not check the other parts of the package config.

TBH I'm not quite sure such a change makes sense. But we should definitely document all breaking changes. I think that we should treat the config properties as a part of the public API.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by dmlloyd**
This was missed in #39295. Fixes #40272.

/cc @mkouba

### CODE REVIEW

**Review by mkouba — COMMENTED** [review]
Oh, I didn't even know about the existence of `ConfigCompatibility` class!

> This is intended to be a temporary and evolutionary solution to be replaced by generative remapping.

I think that it would deserve more attention even though it's a temporary solution.

File: core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigCompatibility.java [hunk 1]
```diff
@@ -55,12 +55,12 @@ public final class ConfigCompatibility {
                     ConfigCompatibility::quarkusPackageIncludedOptionalDependencies),
             entry(List.of("quarkus", "package", "include-dependency-list"),
                     ConfigCompatibility::quarkusPackageIncludeDependencyList),
-            entry(List.of("quarkus", "package", "vineflower", "version"),
-                    ConfigCompatibility::quarkusPackageVineflowerVersion),
-            entry(List.of("quarkus", "package", "vineflower", "enabled"),
-                    ConfigCompatibility::quarkusPackageVineflowerEnabled),
-            entry(List.of("quarkus", "package", "vineflower", "jar-directory"),
-                    ConfigCompatibility::quarkusPackageVineflowerJarDirectory),
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigCompatibility.java:63]: Don't we need to keep these entries and redirect them too?

**Comment by galderz** [comment]
I guess this fixes this right?

```
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:build (default) @ getting-started-reactive ---
[WARNING] [io.quarkus.deployment.configuration] Configuration property 'quarkus.package.vineflower.enabled' has been deprecated and replaced by: [quarkus.package.decompiler.enabled]
[WARNING] [io.quarkus.config] Unrecognized configuration key "quarkus.package.decompiler.enabled" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
```

File: core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigCompatibility.java (hunk 1 above)
Comment (dmlloyd) [reply core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigCompatibility.java:63]: Yes *however* there is some history here - see this series of changes (in reverse chronological order, but the first one is the important one):

- #37445
- #35610
- #25729 / #30909 / #33673

How far should I go to ensure compatibility? I can do any of:

- Just map old Vineflower config
- Map old Vineflower and Quiltflower configs
- Map all of Vineflower, Quiltflower, and Fernflower configs

**Comment by dmlloyd** [comment]
> @dmlloyd slightly related to that, I'm not very happy about how we deprecated the package config properties:
> 
> * the warning is not really actionable: you get a list of various config properties, it's hard to understand what you really need to do to make the warning quiet. I can see how it might not be practical to put all the details there but see below.

I agree to an extent, but other than creating a custom message for each detected usage I wasn't sure how I could do much more. Also imagine that this could be automated some day; how would we want to approach that? Do you want a custom message for each case, and if so, what should the message say? Now, it tells you which property is deprecated, and what properties to set in its place, so at least that gives a good starting point for looking in the docs.

> * we need entries in the migration guide to explain how to fix the setup (especially on how to fix our default project template if you have generated your project pre-3.10 also for the decompiler properties): https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.10

OK, I can look into this.

> * we need something in quarkus-updates to update the projects. I'm going to have a look in the next few days, hopefully in time for the 3.10 release.

OK, I'll see what I can figure out here.

**Comment by dmlloyd** [comment]
> I guess this fixes this right?
> 
> ```
> [INFO] --- quarkus-maven-plugin:999-SNAPSHOT:build (default) @ getting-started-reactive ---
> [WARNING] [io.quarkus.deployment.configuration] Configuration property 'quarkus.package.vineflower.enabled' has been deprecated and replaced by: [quarkus.package.decompiler.enabled]
> [WARNING] [io.quarkus.config] Unrecognized configuration key "quarkus.package.decompiler.enabled" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
> ```

Yes this changes hides the second message.

File: core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigCompatibility.java (hunk 1 above)
Comment (dmlloyd) [reply core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigCompatibility.java]: I went with just mapping the `vineflower` ones and leaving it at that.

**Review by gsmet — APPROVED** [review]

**Pull request merged**